src/named_tuple.cr

```diff
@@ -35,7 +35,22 @@ struct NamedTuple
   # {}             # syntax error
   # ```
   def self.new(**options : **T)
-    options
+    {% if @type.name(generic_args: false) == "NamedTuple" %}
```

Comment:
Note to self: it would be nice to know whether a type is an uninstantiated generic using a macro call.

Reply:
By the way, this works slightly differently from the `Slice.[]` example because it's not possible to attach generic type vars to the macro call directly; they can only be attached via aliases (i.e. `Bytes[...]` is legal while `Slice(UInt8)[...]` isn't).